plugins/dependencies/src/main/kotlin/de/fayard/refreshVersions/BuildSrcLibsTask.kt

```diff
@@ -0,0 +1,139 @@
+        val  configurationsWithHardcodedDependencies = project.findHardcodedDependencies()
+
+        val newEntries = findMissingEntries(configurationsWithHardcodedDependencies)
+
+        writeMissingEntriesInVersionProperties(newEntries)
+        OutputFile.VERSIONS_PROPERTIES.logFileWasModified()
+        Thread.sleep(1000)
+    }
+
+
+
+    @TaskAction
+    fun taskUpdateLibsKt() {
+        val outputDir = project.file(OutputFile.OUTPUTDIR.path)
+
+        val allDependencies = findDependencies()
+        val resolvedUseFqdn = PluginConfig.computeUseFqdnFor(allDependencies, emptyList(), PluginConfig.MEANING_LESS_NAMES)
+        val deps = allDependencies.checkModeAndNames(resolvedUseFqdn)
+
+        val libsFile: FileSpec = kotlinpoet(deps)
+
+        libsFile.writeTo(outputDir)
+        OutputFile.logFileWasModified(OutputFile.LIBS.path, OutputFile.LIBS.existed)
+    }
+
+
+    private fun findDependencies(): List<Dependency> {
+        val allDependencies = mutableListOf<Dependency>()
+        project.allprojects {
+            val projectName = name
+            // println("Configurations: " + configurations.map { it.name })
+            allDependencies += (configurations + buildscript.configurations)
+                //.filter { it.name in PluginConfig.knownConfigurations }
+                .flatMap {
+                    val configurationName = "$projectName:${it.name}"
+                    it.allDependencies.filterIsInstance<ExternalDependency>()
+                        .mapNotNull {
+                            it.group?.let { group ->
+                                Dependency(group, it.name, it.version ?: "none")
+                            }
+                        }
+                }
+        }
+        return allDependencies.distinctBy { d -> d.groupModule() }
+    }
+}
+
+internal fun Project.findHardcodedDependencies(): List<Pair<Project, Configuration>> {
+    val versionsMap = RefreshVersionsConfigHolder.readVersionsMap()
+    val projectsWithHardcodedDependenciesVersions: List<Project> = rootProject.allprojects.filter {
+        it.countDependenciesWithHardcodedVersions(versionsMap) > 0
+    }
+
+    return projectsWithHardcodedDependenciesVersions.flatMap { project ->
+        project.configurations.filterNot { configuration ->
+            configuration.shouldBeIgnored() || 0 == configuration.countDependenciesWithHardcodedVersions(
+                versionsMap = versionsMap,
+                versionKeyReader = RefreshVersionsConfigHolder.versionKeyReader
+            )
+        }.map { configuration -> project to configuration }
+    }
+}
+
+
+internal fun findMissingEntries(configurations: List<Pair<Project, Configuration>>): Map<String, ExternalDependency> {
+
+    val versionsProperties = RefreshVersionsConfigHolder.readVersionsMap()
+
+    val versionKeyReader = RefreshVersionsConfigHolder.versionKeyReader
+
+    val dependencyMap = configurations.flatMap { (project, configuration) ->
```

Comment:
`project` is never used here. Why put it in the configurations list in the first place since it's not used at all?

Reply:
deleted